cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -231,6 +237,9 @@ func (s *service) preStart(c *runc.Container) (handleStarted func(*runc.Containe
 				Container: c,
 				Process:   p,
 			})
+			if init {
+				s.execable[c] = true
```

Comment:
This map is nil

<details>

<summary>panic</summary>

```
panic: assignment to entry in nil map

goroutine 11 [running]:
github.com/containerd/containerd/v2/cmd/containerd-shim-runc-v2/task.(*service).preStart.func1(0xc0000a81e0, {0xd34360, 0xc000190160})
	/home/runner/work/containerd/containerd/cmd/containerd-shim-runc-v2/task/service.go:241 +0x6cb
github.com/containerd/containerd/v2/cmd/containerd-shim-runc-v2/task.(*service).Create(0xc0002001b0, {0xd2e910, 0xc000210180}, 0xc00027c0c0)
	/home/runner/work/containerd/containerd/cmd/containerd-shim-runc-v2/task/service.go:293 +0x3a4
github.com/containerd/containerd/api/runtime/task/v3.RegisterTTRPCTaskService.func2({0xd2e910, 0xc000210180}, 0xc00026c160)
	/home/runner/work/containerd/containerd/vendor/github.com/containerd/containerd/api/runtime/task/v3/shim_ttrpc.pb.go:46 +0x8c
github.com/containerd/ttrpc.defaultServerInterceptor({0xd2e910?, 0xc000210180?}, 0x7f6e6093d108?, 0x10?, 0xc00006af00?)
	/home/runner/work/containerd/containerd/vendor/github.com/containerd/ttrpc/interceptor.go:52 +0x22
github.com/containerd/ttrpc.(*serviceSet).unaryCall(0xc000010618, {0xd2e910, 0xc000210180}, 0xc000010648, 0xc0002682e0, {0xc00018e000, 0x1ea, 0x200})
	/home/runner/work/containerd/containerd/vendor/github.com/containerd/ttrpc/services.go:75 +0xe3
github.com/containerd/ttrpc.(*serviceSet).handle.func1()
	/home/runner/work/containerd/containerd/vendor/github.com/containerd/ttrpc/services.go:118 +0x158
created by github.com/containerd/ttrpc.(*serviceSet).handle in goroutine 10
	/home/runner/work/containerd/containerd/vendor/github.com/containerd/ttrpc/services.go:111 +0x14c
time="2024-08-26T23:36:47.943344153Z" level=info msg="shim disconnected" id=TestRestartMonitor_Always namespace=testing
time="2024-08-26T23:36:47.943374910Z" level=warning msg="cleaning up after shim disconnected" id=TestRestartMonitor_Always namespace=testing
time="2024-08-26T23:36:47.943386833Z" level=info msg="cleaning up dead shim" namespace=testing
time="2024-08-26T23:36:48.065999639Z" level=error msg="copy shim log" error="read /proc/self/fd/14: file already closed" namespace=testing
time="2024-08-26T23:36:48.109368103Z" level=info msg="connecting to shim TestRestartMonitor_Paused_Task" address="unix:///run/containerd/s/6edb8f958c08d0d4617d7d968277d999aaea6d6440c91387f8af295f25a27e84" namespace=testing protocol=ttrpc version=3
```

</details>

Reply:
Serves me right opening a PR without first smoke-testing locally 😅